src/components/application_manager/test/commands/hmi/navi_audio_start_stream_request_test.cc

```diff
@@ -0,0 +1,378 @@
+  EXPECT_CALL(mock_app_manager_, application(kAppID))
+      .WillOnce(Return(mock_app_));
+  (*msg_)[strings::msg_params][strings::app_id] = kAppID;
+
+  command_ = CreateCommand<AudioStartStreamRequest>(msg_);
+
+  EXPECT_CALL(mock_app_manager_, application_by_hmi_app(kAppID))
+      .WillOnce(Return(mock_app_));
+  EXPECT_CALL(*mock_app_, audio_streaming_allowed()).WillOnce(Return(true));
+  EXPECT_CALL(*mock_app_, audio_streaming_approved()).WillOnce(Return(true));
+  const uint32_t retry_number_null = 0u;
+  EXPECT_CALL(*mock_app_, set_audio_stream_retry_number(retry_number_null));
+
+  command_->RetryStartSession();
+}
+
+TEST_F(AudioStartStreamRequestTest,
+       RetryStartSession_AudioStreamNotApproved_UNSUCCESS) {
+  EXPECT_CALL(mock_app_manager_, application(kAppID))
+      .WillOnce(Return(mock_app_));
+  (*msg_)[strings::msg_params][strings::app_id] = kAppID;
+
+  command_ = CreateCommand<AudioStartStreamRequest>(msg_);
+
+  EXPECT_CALL(mock_app_manager_, application_by_hmi_app(kAppID))
+      .WillOnce(Return(mock_app_));
+  EXPECT_CALL(*mock_app_, audio_streaming_allowed()).WillOnce(Return(true));
+  EXPECT_CALL(*mock_app_, audio_streaming_approved()).WillOnce(Return(false));
+  const uint32_t retry_number_null = 0u;
+  EXPECT_CALL(*mock_app_, audio_stream_retry_number())
+      .WillOnce(Return(retry_number_null));
+
+  EXPECT_CALL(mock_message_helper_, SendAudioStartStream(kAppID, _));
+  const uint32_t retry_number_one = 1u;
+  EXPECT_CALL(*mock_app_, set_audio_stream_retry_number(retry_number_one));
+
+  command_->RetryStartSession();
+}
+
+TEST_F(AudioStartStreamRequestTest,
+       RetryStartSession_AudioStreamCanceled_UNSUCCESS) {
+  EXPECT_CALL(mock_app_manager_, application(kAppID))
+      .WillOnce(Return(mock_app_));
+  (*msg_)[strings::msg_params][strings::app_id] = kAppID;
+
+  command_ = CreateCommand<AudioStartStreamRequest>(msg_);
+
+  EXPECT_CALL(mock_app_manager_, application_by_hmi_app(kAppID))
+      .WillOnce(Return(mock_app_));
+  EXPECT_CALL(*mock_app_, audio_streaming_allowed()).WillOnce(Return(true));
+  EXPECT_CALL(*mock_app_, audio_streaming_approved()).WillOnce(Return(false));
+  const uint32_t retry_number = 2u;
+  EXPECT_CALL(*mock_app_, audio_stream_retry_number())
+      .WillOnce(Return(retry_number));
+
+  EXPECT_CALL(mock_app_manager_, EndNaviServices(kAppID));
+
+  command_->RetryStartSession();
+}
+
+TEST_F(AudioStartStreamRequestTest, RetryStartSession_InvalidApp_Canceled) {
+  EXPECT_CALL(mock_app_manager_, application(kAppID))
+      .WillOnce(Return(mock_app_));
+  (*msg_)[strings::msg_params][strings::app_id] = kAppID;
+  command_ = CreateCommand<AudioStartStreamRequest>(msg_);
+
+  EXPECT_CALL(mock_app_manager_, application_by_hmi_app(kAppID))
+      .WillOnce(Return(MockAppPtr()));
+  EXPECT_CALL(*mock_app_, audio_streaming_allowed()).Times(0);
+  EXPECT_CALL(*mock_app_, audio_streaming_approved()).Times(0);
+  const uint32_t retry_number_null = 0u;
```

Comment:
@indlu, you use this kind of variable several times. Perhaps it would be better to move it to a nameless namespace